activerecord/lib/active_record/core.rb

```diff
@@ -601,6 +601,11 @@ def slice(*methods)
       Hash[methods.flatten.map! { |method| [method, public_send(method)] }].with_indifferent_access
     end
 
+    # Returns an array of the values returned by the given methods.
+    def values_at(*methods)
+      methods.flatten.map! { |method| public_send(method) }
```

Comment:
This implementation will raise `NoMethodError` for non-existing methods but `Hash#values_at` returns `nil`.

Is it intended behavior?

Reply:
It makes sense to me because when you call a non-existing method on a model it will raise a `NoMethodError`.

```ruby
Company.first.non_existing_method
#=> NoMethodError (undefined method `non_existing_method' for #<Company:0x00007fe8c747e7c8>)
```

I think we should keep this behavior 👍

Reply:
It is consistent with `#slice` which differs from `Hash#slice` in the same fashion.

Reply:
> It is consistent with #slice which differs from Hash#slice in the same fashion.
Makes sense.